atlasdb-commons/src/main/java/com/palantir/async/initializer/Callback.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.async.initializer;
+
+public abstract class Callback {
```

Comment:
Let's have some javadoc on this class, since we expect other product teams to implement it.

Reply:
Also, (nit) there are already about 15 classes called `Callback`. `InitializationCallback`?